spyder/app/tests/test_mainwindow.py

```diff
@@ -861,11 +863,38 @@ def test_connection_to_external_kernel(main_window, qtbot):
     qtbot.wait(500)
     assert nsb.editor.source_model.rowCount() == 1
 
-    # Shutdown the kernels
-    spykm.stop_restarter()
-    km.stop_restarter()
-    spykm.shutdown_kernel(now=True)
-    km.shutdown_kernel(now=True)
+    # Test runfile in external_kernel
+    run_action = main_window.run_toolbar_actions[0]
+    run_button = main_window.run_toolbar.widgetForAction(run_action)
+
+    # create new file
+    main_window.editor.new()
+    code_editor = main_window.editor.get_focus_widget()
+    code_editor.set_text(
+        "print(2 + 1)"
+    )
+
+    # Start running
+    with qtbot.waitSignal(shell.executed):
+        qtbot.mouseClick(run_button, Qt.LeftButton)
+
+    assert "runfile" in shell._control.toPlainText()
+    assert "3" in shell._control.toPlainText()
+
+    # Try quitting the kernels
+    shell.execute('quit()')
+    python_shell.execute('quit()')
```

Comment:
Is this line necessary? I mean, since `python_shell = shell`, this seems unnecessary to me (and also the declaration `python_shell = shell`).

Reply:
Line 855 is:
` shell = main_window.ipyconsole.get_current_shellwidget()`
so shell != python_shell